File: src/types/safeSite.ts
export type SiteValidationStatus = 
  | 'AI_IDENTIFIED' 
  | 'FIELD_ASSESSED' 
  | 'INFRASTRUCTURE_VALIDATED' 
  | 'COMMUNITY_CONSULTED' 
  | 'APPROVED_FOR_PLANNING';

export type LandVerificationStatus = 
  | 'POTENTIALLY_SUITABLE' 
  | 'NEEDS_LAND_VERIFICATION' 
  | 'OWNERSHIP_VERIFICATION_REQUIRED' 
  | 'ADMINISTRATIVE_REVIEW_REQUIRED';

export interface CarryingCapacityBreakdown {
  physicalLandCapacity: number; // Max population based on sq m/person
  waterSupportedCapacity: number;
  infrastructureCapacity: number;
  healthcareCapacity: number;
  educationCapacity: number;
  roadNetworkCapacity: number;
  recommendedMaxCapacity: number; // Constrained by bottleneck
  currentOccupancy?: number;
  remainingCapacity?: number;
  bottleneckResource: string;
  isOverCapacity?: boolean;
}

export interface InfrastructureGap {
  category: 'HOUSING' | 'WATER' | 'HEALTHCARE' | 'EDUCATION' | 'POWER' | 'ROADS' | 'COMMUNICATION';
  required: string;
  available: string;
  gapStatus: 'SATISFIED' | 'MODERATE_GAP' | 'CRITICAL_GAP';
  estimatedCostCr: number;
  hasDataGap?: boolean;
}

export interface CommunityAcceptanceMetrics {
  overallScore: number; // 0 - 100
  distanceScore: number;
  livelihoodContinuityScore: number; // Fishing / Agriculture / Industrial
  culturalCompatibilityScore: number;
  schoolHealthcareScore: number;
  socialNetworkScore: number;
  livelihoodType: 'COASTAL_FISHING' | 'AGRICULTURE' | 'AGRI_ALLIED' | 'SERVICES_URBAN';
  primaryConcerns: string[];
}

export interface SafeSite {
  id: string;
  name: string;
  code: string;
  district: string;
  state: string;
  latitude: number;
  longitude: number;
  landAreaAcres: number;
  currentLandUse: string;
  elevationMeters: number;
  distanceFromCoastKm: number;
  slopePercentage: number;
  
  // Status & Scores
  status: SiteValidationStatus;
  landVerificationStatus?: LandVerificationStatus;
  overallScore: number; // 0 - 100
  hazardSafetyScore: number; // 0 - 100
  landAvailabilityScore: number;
  connectivityScore: number;
  waterScore: number;
  healthcareScore: number;
  educationScore: number;
  livelihoodScore: number;
  administrativeFeasibilityScore?: number;
  
  // Climate Horizon
  climateHorizonYears?: number; // e.g. 50 years zero-flood envelope
  seaLevelRiseBufferMeters?: number;
  
  // Carrying Capacity
  capacity: CarryingCapacityBreakdown;
  allocatedPopulation: number;
  
  // Community Acceptance
  communityAcceptance: CommunityAcceptanceMetrics;
  
  // Infrastructure Specs
  infrastructureGaps: InfrastructureGap[];
  estimatedDevelopmentCostCr: number;
  estimatedDevelopmentTimeMonths: number;
  
  // Distance / Accessibility from target settlements
  distanceFromKeySettlementKm: { [settlementId: string]: number };
  travelTimeMin: { [settlementId: string]: number };
  roadQuality: 'EXCELLENT_HIGHWAY' | 'PAVED_ALL_WEATHER' | 'RURAL_UPGRADE_NEEDED';
  
  // Environmental Considerations
  ecologicalSensitivity: 'LOW' | 'MODERATE' | 'HIGH';
  environmentalClearanceRisk: 'MINIMAL' | 'REQUIRES_CRZ_CLEARANCE' | 'STANDARD_PERMITS';
  
  // Data Gap Tracking
  dataConfidenceScore?: number;
  hasUnverifiedLandStatus?: boolean;
  
  description: string;
  selectionRationale: string;
}

export interface SiteMatchResult {
  site: SafeSite;
  matchScore: number;
  safetyScore: number;
  capacityScore: number;
  accessibilityScore: number;
  communityAcceptanceScore: number;
  livelihoodCompatibilityScore: number;
  distanceKm: number;
  travelTimeMin: number;
  isRecommended: boolean;
  recommendationReason: string;
  carryingCapacityHeadroom?: number;
  whyRecommendedPoints?: string[];
}

File: backend/app/models/safe_site.py
from sqlalchemy import Column, String, Integer, Float, Boolean, JSON, Text
from app.core.database import Base


class SafeSiteModel(Base):
    __tablename__ = "safe_sites"

    id = Column(String(64), primary_key=True, index=True)
    name = Column(String(128), nullable=False, index=True)
    code = Column(String(64), nullable=False)
    district = Column(String(128), nullable=False, index=True)
    state = Column(String(64), default="Tamil Nadu")
    latitude = Column(Float, nullable=False)
    longitude = Column(Float, nullable=False)
    land_area_acres = Column(Float, default=10.0)
    current_land_use = Column(String(128), default="Government Wasteland (Gramanatham)")
    elevation_meters = Column(Float, default=15.0)
    distance_from_coast_km = Column(Float, default=10.0)
    slope_percentage = Column(Float, default=2.0)

    # Status & Suitability Scores
    status = Column(String(64), default="AI_IDENTIFIED")
    land_verification_status = Column(String(64), default="POTENTIALLY_SUITABLE")
    overall_score = Column(Integer, default=80)
    hazard_safety_score = Column(Integer, default=90)
    land_availability_score = Column(Integer, default=85)
    connectivity_score = Column(Integer, default=80)
    water_score = Column(Integer, default=75)
    healthcare_score = Column(Integer, default=70)
    education_score = Column(Integer, default=70)
    livelihood_score = Column(Integer, default=80)

    # Carrying Capacity
    capacity_recommended_max = Column(Integer, default=3000)
    allocated_population = Column(Integer, default=0)
    bottleneck_resource = Column(String(128), default="Potable Water Distribution")

    # Community Acceptance
    community_acceptance_score = Column(Integer, default=80)
    livelihood_type = Column(String(64), default="COASTAL_FISHING")

    # Financials & Time
    estimated_development_cost_cr = Column(Float, default=15.0)
    estimated_development_time_months = Column(Integer, default=18)

    # Distance / Travel time from key settlements
    distance_from_key_settlements = Column(JSON, nullable=True)  # {"kadalpuram": 18.4, "malaiyur": 45.0}
    travel_time_min = Column(JSON, nullable=True)               # {"kadalpuram": 32, "malaiyur": 75}
    road_quality = Column(String(64), default="PAVED_ALL_WEATHER")

    description = Column(Text, nullable=True)
    selection_rationale = Column(Text, nullable=True)
    data_classification = Column(String(64), default="DEMO / SYNTHETIC DATA")
